enhancements/machine-config/install-time-support-for-image-mode-on-openshift.md

```diff
@@ -0,0 +1,343 @@
+2. MCO creates MachineOSConfig objects in the Kubernetes API if they don't already exist
+3. Build controller detects the `machineconfiguration.openshift.io/pre-built-image` annotation on the newly created MachineOSConfig
+4. Build controller triggers seeding workflow by calling `seedMachineOSConfigWithExistingImage()``
+5. MCO creates a synthetic MachineOSBuild with:
+    - Successful status referencing the seeded pre-built OS container image
+    - `PreBuiltImage` label set to "true"
+    - Proper metadata including MOSC name, DigestedImagePushSpec, and MCP reference
+    - Success condition with reason "PreBuiltImageSeeded"
+6. MCO updates MachineOSConfig status with `CurrentImagePullSpec` pointing to the pre-built image
+    - This creates consistency between the actual node state (already running pre-built image) and the MCO API objects for future OCL workflows
+
+#### Step 4: Post-Install Integration
+
+1. Any future updates/changes will use the current post-install workflow to trigger a new image build and roll it out onto the nodes based on the MachineOSConfig created at install time
+
+#### Notes and Restrictions:
+
+- A pre-built custom OS container image is needed. Image build during bootstrap is not supported
+- The Containerfile used to build the image needs to be added to the MachineOSConfig CRs created at install time
+- Two boots will occur: Initial disk image boot with stock RHCOS image and a second boot when the MCD first boot service runs to boot into the custom OS container image
+- The registry where the pre-built image is stored will need to be accessible to the cluster in future as any future builds will be pushed there and pulled from there when rolling out onto the nodes
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+We currently don't have support for Hypershift with post-install Image Mode. Install time support for this will be added (if needed) once we have post-install support for it. This is currently not part of the implementation plan for this enhancement.
+
+#### Standalone Clusters
+
+Clusters in disconnected environments require special environment configurations such as a local package mirror, local image registry, etc.
+Post-install Image Mode currently works for disconnected environments and should work in a similar manner for install time
+
+#### Single-node Deployments or MicroShift
+
+Single-Node OpenShift (SNO) currently works fine with post-install Image Mode and should continue to work as-is for install time support as well.
+
+### API Extensions
+
+None
+
+### Implementation Details/Notes/Constraints
+
+#### Bootstrap Process Integration
+
+The bootstrap MCO will be enhanced to:
+
+- Parse MOSC configurations from the installation directory
+- Update the MCD first boot service to use the seeded pre-built custom OS container image
+
+#### Node Deployment Process
```

Comment:
This is also the same as we have today right (for non layered pools)?

Reply:
yes, it is the same as we have today for the non-ocl path when using the osImageURL MC